docs/creating-releases.md

```diff
@@ -0,0 +1,141 @@
+   1. Update [`src/LICENSE.txt`][licensetxt], retaining the content to the first
+      horizontal rule, which is the license for PR Metrics itself. All content
+      following this line should be replaced with the downloaded licenses. Note
+      that the ordering of licenses may change.
+   1. If the download dialog includes any notice indicating that license
+      information could not be located at [Clearly Defined][clearlydefined], you
+      will need to add the information to that source. To do this, expand the
+      drop down menu in the dialog to reveal the problematic dependencies. For
+      each dependency:
+
+      1. Navigate to the Clearly Defined [Harvest page][clearlydefinedharvest].
+      1. In the first search box, select "NpmJS".
+      1. In the second search box, select the dependency name.
+      1. In the "Pick an npm version" box, select the appropriate version.
+      1. Click "Harvest".
+
+      It is also possible to add all dependencies to the page and click
+      "Harvest" afterwards to harvest all license information simultaneously.
+
+      Wait some time for harvesting to complete and try regenerating the license
+      information at the Component Governance page. Repeat the process until all
+      license information is available.
+1. Commit all the changes to your branch, updating the PR.
+1. Ensure all build stages continue to run successfully
+1. Commit the changes to the `main` branch.
+1. Wait for the [`main` branch build loop][mainbuild] to complete successfully.
+
+### Creating the Release
+
+1. Open the [Releases][releases] page.
+1. Click "Draft a new release".
+1. Enter the new version number in the format of "vX.X.X" in the "Choose a tag"
+   field, replacing the "X.X.X" with the version number. Click "Create new tag:
+   vX.X.X on publish".
+1. Enter "Release vX.X.X" in the "Release title" field, replacing the "X.X.X"
+   with the version number.
+1. Click "Generate release notes".
+1. At the top of the "Describe this release" field, add "## Summary" and a
+   bulleted list of the major changes since the last release.
+1. Enable "Create a discussion for this release" and select category "Releases".
+1. Click "Publish release".
+1. Wait for the [Release build loop][releasebuild] to complete successfully.
+1. Click into the completed build and download the `ms-omex.PRMetrics` artifact.
+1. Extract the .vsix file from the compressed artifact.
+1. Go to the Visual Studio Marketplace page for the
+   [`ms-omex` publisher][marketplace]. You will need to have appropriate
+   permissions to access this page.
+1. Click "..." next "PR Metrics" and click "Update".
+1. Follow the instructions to upload .vsix file.
+1. Click "Upload".
+1. Wait for validation to complete and the new version of the extension to be
+   published.
+
+## Security Updates
+
+If Dependabot opens an out-of-band security fix PR, complete the PR and follow
+the [Quarterly Releases](#quarterly-releases) steps.
```

Comment:
Should this only be applied for Medium/Critical or above fixes?

Reply:
I'd favour releasing with the fix if the PR is automatically opened, as the existence of the PR on GitHub might help point attackers to a potential attack surface. It isn't as urgent in low priority cases of course, but there is still some risk, so it's better to close it ASAP.

The release process in general isn't too involved. It looks like a lot of steps, but it's fairly quick to run through in reality and it tends to be very safe, so I'd favour erring on the side of caution.